sympy/matrices/expressions/tests/test_indexing.py

```diff
@@ -259,11 +259,18 @@ def test_matrix_expression_from_index_summation():
     assert MatrixExpr.from_index_summation(expr, a) == A*B
 
     expr = Sum(KroneckerDelta(i1, m)*KroneckerDelta(i2, n)*A[i, i1]*A[j, i2], (i1, 0, k-1), (i2, 0, k-1))
-    assert MatrixExpr.from_index_summation(expr, m) == A.T*A[j, n]
 
     # Test numbered indices:
     expr = Sum(A[i1, i2]*w1[i2, 0], (i2, 0, k-1))
     assert MatrixExpr.from_index_summation(expr, i1) == A*w1
 
     expr = Sum(A[i1, i2]*B[i2, 0], (i2, 0, k-1))
     assert MatrixExpr.from_index_summation(expr, i1) == MatrixElement(A*B, i1, 0)
+
+
+@XFAIL
+def test_kronecker_delta_from_index_summation():
```

Comment:
@Upabjojr if you know how to fix this then it would be good to do that. Otherwise I think it is better to XFAIL this for now.

I would like to put out a release soon and I want this to be fixed in that release.

Reply:
Well, this code is going to be replaced by two functions in the module `sympy.tensor.array.expressions`. Specifically, `convert_indexed_to_array( )` and `convert_array_to_matrix( )`. At the current state I'm not sure they already support all of this, but once they will, `MatrixExpr.from_index_summation` will have to be deprecated.